### PR TITLE
Remove SurveyNotice in AppBanner

### LINKED ISSUE #10166: Remove SurveyNotice in AppBanner
Let's remove `SurveyNotice` in `AppBanner`. We don't use this component and it
gets included in the final bundle. This won't have a massive impact but it's
useless.

### PR DESCRIPTION
**Pull request by willdurand**
Fixes #10166

---

From 435.43KB to 434.28KB according to `bundlesize` (it's not a lot but we don't need it so...)

### CODE REVIEW

File: src/amo/components/SurveyNotice/index.js [hunk 1]
```diff
@@ -39,6 +39,9 @@ type InternalProps = {|
   wasDismissed: boolean,
 |};
 
+// Important: we don't use this component right now. It was removed from
+// `AppBanner` in https://github.com/mozilla/addons-frontend/pull/10129
+// because it wasn't needed.
```
Comment (eviljeff) [inline src/amo/components/SurveyNotice/index.js:44]: If we're not using the component, delete it?  Unused code starts to smell.

**Review by eviljeff — APPROVED** [review]
The effort (and the counter-effort to revert that would be needed when/if we need SurveyNotice in the future) doesn't seem justified by 0.5% bundle size reduction, imo.  (Anything less than 5% reduction is dubious)

File: src/amo/components/SurveyNotice/index.js (hunk 1 above)
Comment (willdurand) [reply src/amo/components/SurveyNotice/index.js:44]: I agree but @bobsilverberg wanted to keep it..

Comment (bobsilverberg) [reply src/amo/components/SurveyNotice/index.js:44]: I actually never said that exactly. I was just debating the merits of keeping it vs deleting it.

Comment (willdurand) [reply src/amo/components/SurveyNotice/index.js:44]: @bobsilverberg are you ok to delete the component then?

Comment (bobsilverberg) [reply src/amo/components/SurveyNotice/index.js:44]: Yes

**Pull request merged**